src/SFA.DAS.ApprenticeFeedback.Application/Commands/GenerateFeedbackTransactions/GenerateFeedbackTransactionsCommandHandler.cs

```diff
@@ -0,0 +1,43 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+using MediatR;
+using SFA.DAS.ApprenticeFeedback.Domain.Configuration;
+using SFA.DAS.ApprenticeFeedback.Domain.Entities;
+using SFA.DAS.ApprenticeFeedback.Domain.Interfaces;
+using static SFA.DAS.ApprenticeFeedback.Domain.Models.Enums;
+
+namespace SFA.DAS.ApprenticeFeedback.Application.Commands.GenerateFeedbackTransactions
+{
+    public class GenerateFeedbackTransactionsCommandHandler : IRequestHandler<GenerateFeedbackTransactionsCommand, GenerateFeedbackTransactionsCommandResponse>
+    {
+        private readonly IFeedbackTransactionContext _transactionContext;
+        private readonly IApprenticeFeedbackTargetContext _targetContext;
+        private readonly ApplicationSettings _settings;
+
+        public GenerateFeedbackTransactionsCommandHandler(IFeedbackTransactionContext transactionContext, IApprenticeFeedbackTargetContext targetContext, ApplicationSettings settings)
+        {
+            _transactionContext = transactionContext;
+            _targetContext = targetContext;
+            _settings = settings;
+        }
+
+        public async Task<GenerateFeedbackTransactionsCommandResponse> Handle(GenerateFeedbackTransactionsCommand request, CancellationToken cancellationToken)
+        {
+            IEnumerable<GenerateFeedbackTransactionsResult> result = await _transactionContext.GenerateFeedbackTransactionsAsync();
+
+            if (result == null || !result.Any())
```

Comment:
Any reason for not just always returning the `GenerateFeedbackTransactionsCommandResponse` with a Count of 0 with no CreatedOn value or set to null?

Reply:
That's what it does, the result is the rows returned by the stored proc. There should always be one. If nothing has been inserted it will return a row with the count = 0.
This code catches the unexpected case that no row of results is returned by the stored proc.

Reply:
Ah, yes I see, And agree!
Changed as requested.